Improve how to deploy a collection (#172)
OpenTermsArchive/docs: content/deployment/how-to/deploy.md

# How to deploy a collection

This guide will help you deploy an Open Terms Archive collection to a server. The deployment is automated using [Ansible](https://docs.ansible.com/ansible/latest/index.html) and will set up the Open Terms Archive engine and configure it to track your collection's terms.

> ## System Overview

Reviewer: ```suggestion
## System overview
```

Before diving into the deployment steps, here's a quick reminder of the high-level overview of how Open Terms Archive works. 

Each collection uses three repositories, `declarations` which contains the terms to track, the engine and deployment configuration, `versions` and `snapshots` which are automatically managed repositories storing versions and snapshots history.

The deployment process is automated through GitHub Actions. Ansible configures the server and sets up the Open Terms Archive engine. On the server, [PM2](https://pm2.keymetrics.io) is used to start and control the engine.

> When deployed, the engine runs continuously on the server, periodically checking for changes in the tracked terms. When changes are detected, it automatically commits them to the versions and snapshots repositories. When issues occur, email notifications are sent.

Reviewer: Shoudln't this be a reference or an explanation?

## Prerequisites

Before starting, ensure you have:

- A server with admin access
- All collections repositories created, if not, see the [guide to create repositories]({{< relref "collections/how-to/create-repositories" >}})

> - At least one declaration added to your collection

Reviewer: ```suggestion
- At least one declaration in your collection (if you created your declaration from the Demo template, one is provided by default)
```

Author: When we use the demo template, all declarations are removed by the first time setup process.

> - A GitHub user account dedicated to bot-related actions (commit entries in versions and snapshots repositories, report issues when tracking fails, publish releases, …)

Reviewer: ```suggestion
- A GitHub user account to automate actions such as committing entries in versions and snapshots repositories, reporting issues when tracking fails, publishing releases…
```

## 1. Configure the server

First, ensure your server provides unsupervised access:

1. Check the SSH host key and get the SSH fingerprint by running the following command on your local machine:

   ```shell
   ssh-keyscan -t ed25519 <server_address>
   ```

   If no Ed25519 key appears, generate one by running the following commands on the server:

   ```shell
   sudo ssh-keygen -t ed25519 -f /etc/ssh/ssh_host_ed25519_key
   sudo systemctl restart ssh
   ```

   > **Note**: A server fingerprint is a unique identifier for your server's SSH key. It helps verify that you're connecting to the correct server and not a malicious one. The fingerprint is a hash of the server's public key and is used to prevent man-in-the-middle attacks. You'll need this fingerprint in the next steps for secure deployment.

2. Create a dedicated user account specifically for deployment purposes, by running the following commands on the server:

   ```shell
   adduser <deployment_user>
   usermod --append --groups=sudo <deployment_user>
   ```

   > **Note**: The `adduser` command might not be installed by default on your system. It can be installed with `sudo apt-get install adduser`.

> 3. Configure passwordless sudo access for this user, by running the adding the following line to the `/etc/sudoers` file on the server:

Reviewer: ```suggestion
3. Configure passwordless sudo access for this user, by adding the following line to the `/etc/sudoers` file on the server:
```

## 7. Configure email notifications

This section describes how to configure the engine to use a specific SMTP server to send email notifications when it encounters errors during the tracking process. This helps you stay informed about issues that need attention and allows you to restart the tracking process if needed.

1. Get the SMTP credentials (host, username, password) from your email provider

2. Update collection SMTP configuration within the `logger` key of `@opentermsarchive/engine` in the `config/production.json` file:

   ```json
      "logger": {
         "smtp": {
            "host": "<smtp_host>",
            "username": "<smtp_username>"
         },
      },
   ```

3. Store the password in `deployment/.env`:

   ```shell
   OTA_ENGINE_SMTP_PASSWORD=your_smtp_key
   ```

   > **Note**: To decrypt the file encrypted in a previous step in order to add the password, run `ansible-vault decrypt .env`

4. Encrypt the `.env` file:

   ```shell
   ansible-vault encrypt .env
   ```

5. > _Specific to Open Terms Archive organization members_
   > Create a new SMTP key in Brevo and name it "<collection_name> collection"

>    > Back up the key in the shared password database by creating an entry titled "SMTP Key" in the collection folder and storing the credentials in this entry

Reviewer: Isn't this part obsolete?

Author: No, it is used for sending error emails
